Explorer/Assets/Scripts/CrdtEcsBridge/JsModulesImplementation/ClientWebSocketApiImplementation.cs

```diff
@@ -0,0 +1,225 @@
+﻿using CrdtEcsBridge.PoolsProviders;
+using Cysharp.Threading.Tasks;
+using Microsoft.ClearScript.JavaScript;
+using SceneRuntime;
+using SceneRuntime.Apis.Modules;
+using System;
+using System.Collections.Concurrent;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CrdtEcsBridge.JsModulesImplementation
+{
+    /// <summary>
+    /// Uses raw .NET ClientWebSocket under the hood
+    /// </summary>
+    public class ClientWebSocketApiImplementation : IWebSocketApi
+    {
+        private static readonly ChunkTransmission CHUNK_TRANSMISSION = new ();
+
+        private readonly IInstancePoolsProvider instancePoolsProvider;
+        private readonly IJsOperations jsOperations;
+
+        private readonly ConcurrentDictionary<int, WebSocketRental> webSockets = new ();
+
+        private int nextId;
+
+        public ClientWebSocketApiImplementation(IInstancePoolsProvider instancePoolsProvider, IJsOperations jsOperations)
+        {
+            this.instancePoolsProvider = instancePoolsProvider;
+            this.jsOperations = jsOperations;
+        }
+
+        public void Dispose()
+        {
+            foreach (WebSocketRental rental in webSockets.Values)
+                rental.Dispose();
+
+            webSockets.Clear();
+        }
+
+        public int CreateWebSocket(string url)
+        {
+            Interlocked.Increment(ref nextId);
+
+            webSockets[nextId] = new WebSocketRental(); // ClientWebSocket does not support reviving
+            return nextId;
+        }
+
+        public async UniTask ConnectAsync(int websocketId, string url, CancellationToken ct)
+        {
+            await GetInstanceOrThrow(websocketId).WebSocket.ConnectAsync(new Uri(url), ct);
+        }
+
+        public async UniTask SendBinaryAsync(int websocketId, IArrayBuffer data, CancellationToken ct)
+        {
+            WebSocketRental webSocket = GetInstanceOrThrow(websocketId);
+
+            var bytesCount = (int)data.Size;
+            if (bytesCount == 0) return;
+
+            using PoolableByteArray poolableArray = instancePoolsProvider.GetCrdtRawDataPool(bytesCount);
```

Comment:
Is this the CrdtRawDataPool being used in a non-crdt context? So it is more generic? Maybe we can just rename it for clarity?